macro/eventDisplay.py

```
@@ -1172,9 +1172,14 @@ def debugStraw(n) -> None:
 
 inFile = ROOT.FairFileSource(options.InputFile)
 fRun.SetSource(inFile)
+ROOT.SetOwnership(inFile, False)  # C++ FairRun takes ownership
 if options.OutputFile is None:
-    options.OutputFile = ROOT.TMemFile("event_display_output", "recreate")
-fRun.SetSink(ROOT.FairRootFileSink(options.OutputFile))
+    import tempfile
+
+    options.OutputFile = tempfile.mktemp(suffix=".root")
```

Ruff (0.15.13)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)